app/src/main/java/org/jellyfin/androidtv/ui/startup/StartupActivity.kt

```diff
@@ -92,64 +93,51 @@ class StartupActivity : FragmentActivity() {
 		}
 	}
 
-	fun openNextActivity() {
-		val itemId = intent.getStringExtra(ITEM_ID)
-		val itemIsUserView = intent.getBooleanExtra(ITEM_IS_USER_VIEW, false)
+	private suspend fun openNextActivity() {
+		val itemId = intent.getStringExtra(EXTRA_ITEM_ID)
+		val itemIsUserView = intent.getBooleanExtra(EXTRA_ITEM_IS_USER_VIEW, false)
+
 		if (itemId != null) {
 			if (itemIsUserView) {
-				apiClient.GetItemAsync(itemId, apiClient.currentUserId, object : Response<BaseItemDto?>() {
-					override fun onResponse(item: BaseItemDto?) {
-						ItemLauncher.launchUserView(item, this@StartupActivity, true)
-					}
+				// Try opening the user view
+				val item = callApi<BaseItemDto?> { apiClient.GetItemAsync(itemId, apiClient.currentUserId, it) }
 
-					override fun onError(exception: Exception) {
-						// go straight into last connection
-						val intent = Intent(application, MainActivity::class.java)
-						startActivity(intent)
-						finish()
-					}
-				})
+				if (item != null) {
+					ItemLauncher.launchUserView(item, this, true)
+					finish()
```

Comment:
I want to use `finishAfterTransition` here but the itemlauncher does async magic so that's not possible right now.